src/transactions/TransactionFrame.cpp

```diff
@@ -1170,11 +1176,15 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
             }
 
             outerMeta.pushContractEvents(std::move(mEvents));
+            outerMeta.pushDiagnosticEvents(std::move(mDiagnosticEvents));
 #endif
         }
         else
         {
             markResultFailed();
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+            outerMeta.pushDiagnosticEvents(std::move(mDiagnosticEvents));
```

Comment:
nit: move is redundant here (unless these functions take an rvalue reference)

Reply:
`TransactionMetaFrame` takes the events as rvalue references.